src/app.js

```diff
@@ -42,38 +42,31 @@ app.get('/products', (req, res) => {
     })
 })
 
-app.get('/weather', (req, res) => {
+app.get('/weather', async (req, res) => {
     if (!req.query.address){
         return res.send({
             error: 'You must provide an address.'
         })
     }
 
-    geocode(req.query.address, mapboxKey, (error, {latitude, longitude, location} = {}) => {
-        if (error){
-            return res.send({error})
-        }
-
-        forecast(latitude, longitude, darkskyKey, (error, forecastData) => {
-            if (error) {
-                return res.send({error})
-            }
-
-            res.send({
-                forecast: forecastData,
-                location,
-                address: req.query.address
-            })
+    try {
+        const { latitude, longitude, location } = await geocode(req.query.address, mapboxKey)
+        const forecastData = await forecast(latitude, longitude, darkskyKey)
+        res.send({
+            forecast: forecastData,
+            location,
+            address: req.query.address
         })
-    })
-   
+    } catch(error) {
+        res.send({ error })
+    }
 })
 
 
 app.get('*', (req, res) => {
     res.render('404', {
-        title: 'Opps, something\'s wrong.',
-        msg: 'This doesn\'t exist, try another page?',
+        title: "Oops, something's wrong.",
```

Comment:
if I'm used to writing single quote, but I need to use \, is it always better to switch to double quote? (for clean code principle?)

Reply:
I generally like to use double quotes when I have single quotes inside the string. I just think it is easier to read. It really doesn't matter, though. I just thought I would change it incase you didn't know you could do that.

Reply:
Nice, I like this way better. It's more human-readable.